tests/test_daemon.cpp

```diff
@@ -532,10 +532,10 @@ MATCHER_P(YAMLSequenceContainsStringMap, values, "")
             continue;
         for (auto it = values.cbegin();; ++it)
         {
-            if (node[it->first].template as<std::string>() != it->second)
-                break;
-            else if (it == values.cend())
+            if (it == values.cend())
                 return true;
+            else if (node[it->first].template as<std::string>() != it->second)
+                break;
```

Comment:
Hey, can you briefly explain why this change? I can't immediately see the difference?

Reply:
This has been a bug all along, but only exposed by the changes in the updated `yaml-cpp`.  We really want to check if `it == cend()` first, because if it is and using the previous code, then `it->first` is an invalid access.  With the new `yaml-cpp`, the previous code would result in a segfault.

Reply:
D'uh, of course :)